src/main/java/org/jabref/logic/ai/AiChat.java

```diff
@@ -0,0 +1,101 @@
+ */
+public class AiChat {
+    // Stores the embeddings and the chat history.
+    private final AiChatData data;
+    // The main class that executes user prompts. Maintains API calls and retrieval augmented generation (RAG).
+    private final ConversationalRetrievalChain chain;
+    // TODO: It turns out ChatMemoryStore is a more global class than I though. Make it global.
+    // Initially I thought it is local to every chat.
+    // But no, ChatMemoryStore is a store for many different chats.
+    // In current implementation I use it as I said it first: different ChatMemoryStore for different chats,
+    // so I need ID for several algorithms.
+    private final Object chatId;
+
+    public AiChat(AiConnection aiConnection) {
+        this.data = new AiChatData();
+
+        // This class is basically an "algorithm class" for retrieving the relevant contents of documents.
+        ContentRetriever contentRetriever = EmbeddingStoreContentRetriever
+                .builder()
+                .embeddingStore(data.getEmbeddingStore())
+                .embeddingModel(aiConnection.getEmbeddingModel())
+                .build();
+
+        // This class is also an "algorithm class" that maintains the chat history.
+        // An algorithm for managing chat history is needed because you cannot stuff the whole history for the AI:
+        // there would be too many tokens. This class, for example, sends only the 10 last messages.
+        ChatMemory chatMemory = MessageWindowChatMemory
+                .builder()
+                .chatMemoryStore(data.getChatMemoryStore())
+                .maxMessages(10) // This was the default value in the original implementation.
+                .build();
+
+        this.chatId = chatMemory.id();
+
+        // TODO: Investigate whether the ChatMemoryStore holds the whole chat history.
+        // As I think, ChatMemory should only send 10 last messages to the AI provider, but not remove the last
+        // messages from the store.
+
+        this.chain = ConversationalRetrievalChain
+                .builder()
+                .chatLanguageModel(aiConnection.getChatModel())
+                .contentRetriever(contentRetriever)
+                .chatMemory(chatMemory)
+                .build();
+    }
+
+    public AiChat(AiChatData data, AiConnection aiConnection) {
+        // Code duplication...
+
+        this.data = data;
+
+        ContentRetriever contentRetriever = EmbeddingStoreContentRetriever
+                .builder()
+                .embeddingStore(data.getEmbeddingStore())
+                .embeddingModel(aiConnection.getEmbeddingModel())
+                .build();
+
+        ChatMemory chatMemory = MessageWindowChatMemory
+                .builder()
+                .chatMemoryStore(data.getChatMemoryStore())
+                .maxMessages(10) // This was the default value in the original implementation.
```

Comment:
Extract a constant (Ctrl+Alt+F). Reason: In general, all numbers should be constants (and there should not be magic numbers in the code)